WPMedia/fusion-news-theme-blocks
# updated cardlist block to respect offsetOverride

**Be sure to run `npm test`, `npm run lint`, and write detailed test steps before requesting reviewers**

## Description

#### Jira Ticket: [THEMES-1951](https://arcpublishing.atlassian.net/browse/THEMES-1951)

Previously, only the headline would update when you added an offset Override. The image, date and attribution all stayed the same (from the first story). This update makes it so that all data and the image is accurate based on the offset Override. This doc has more info on the proper configuration of card-list-blocks: https://docs.arcxp.com/alc/en/themes-using-promo-and-list-blocks?sys_kb_id=0ef1355483a202d0107cb2b8beaad3e0&id=kb_article_view&sysparm_rank=1&sysparm_tsqueryId=2890afec83e282d0107cb2b8beaad3b4#mcetoc_1heiur8ipb

Before:
<img width="1541" alt="image" src="https://github.com/WPMedia/arc-themes-blocks/assets/85515364/2ee01784-785c-4bbc-8bbe-6b897ee00bd5">
After: 
<img width="1524" alt="image" src="https://github.com/WPMedia/arc-themes-blocks/assets/85515364/1fbec341-f781-4f72-b29d-2c062edbfc31">


## Test Steps

_Add detailed test steps a reviewer must complete to test this PR_

1. Checkout this branch `git checkout THEMES-1951`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/card-list-block`
3. Add 3 card list blocks to a page (I added to a triple chain to make them easier to compare)
4. In each block, select a collection as the content source, and use the configuration below with overrides of 0,1, and 2:
<img width="330" alt="image" src="https://github.com/WPMedia/arc-themes-blocks/assets/85515364/a747117d-46b1-48df-b1bc-807a5a185106">
5. Make sure that the content rendered in each block corresponds to the correct story in the collection.

[THEMES-1951]: https://arcpublishing.atlassian.net/browse/THEMES-1951?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
